viamonvif/discovery.go

```diff
@@ -1,241 +1,120 @@
 // Package viamonvif provides ONVIF integration to the viamrtsp module
+//nolint: revive
 package viamonvif
 
 import (
-	"bytes"
 	"context"
 	"encoding/xml"
-	"errors"
 	"fmt"
-	"io"
 	"net"
-	"net/http"
 	"net/url"
-	"slices"
 	"strings"
-	"sync"
 
-	"github.com/use-go/onvif"
-	"github.com/use-go/onvif/device"
-	"github.com/use-go/onvif/media"
-	wsdiscovery "github.com/use-go/onvif/ws-discovery"
-	onvifxsd "github.com/use-go/onvif/xsd/onvif"
+	"github.com/viam-modules/viamrtsp/viamonvif/device"
+	"github.com/viam-modules/viamrtsp/viamonvif/xsd/onvif"
 	"go.viam.com/rdk/logging"
-	"go.viam.com/utils"
-)
-
-const (
-	streamTypeRTPUnicast = "RTP-Unicast"
 )
 
 // OnvifDevice is an interface to abstract device methods used in the code.
 // Used instead of onvif.Device to allow for mocking in tests.
 type OnvifDevice interface {
-	CallMethod(request interface{}) (*http.Response, error)
+	GetDeviceInformation() (device.GetDeviceInformationResponse, error)
+	GetProfiles() (device.GetProfilesResponse, error)
+	GetStreamURI(profile onvif.Profile, creds device.Credentials) (*url.URL, error)
 }
 
-// callAndParse sends a request to an ONVIF device, decoding and mutating the
-// provided response struct with the result. Returns an error on failure.
-func callAndParse(logger logging.Logger, deviceInstance OnvifDevice, request interface{}, response any) error {
-	resp, err := deviceInstance.CallMethod(request)
-	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
+// DiscoverCameras performs WS-Discovery
+// then uses ONVIF queries to get available RTSP addresses and supplementary info.
+func DiscoverCameras(creds []device.Credentials, manualXAddrs []*url.URL, logger logging.Logger) (*CameraInfoList, error) {
```

Comment:
I also changed the structure of this funtion. 

It will now:
1. Run WSDiscover
2. Take the urls returned from WSDiscover + any provided by the caller as well as the list of credentials and attempt to query device information from those host's onvif endpoints.
3. Add the succesful onvif device information (rtsp_urls, host, model, etc) to the response